tests/conftest.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from typing import Generator
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from nowcasting_datamodel.models.base import Base_Forecast
+from nowcasting_datamodel.models import MLModelSQL
+
+# Shared Test Configuration Constants
+TEST_DB_URL = "postgresql://postgres:12345@localhost/testdb"
+RESOURCE_ID = "db6c038f-98af-4570-ab60-24d71ebd0ae5"
+LIMIT = 5
+COLUMNS = ["DATE_GMT", "TIME_GMT", "EMBEDDED_SOLAR_FORECAST"]
+RENAME_COLUMNS = {
+    "DATE_GMT": "start_utc",
+    "TIME_GMT": "end_utc",
+    "EMBEDDED_SOLAR_FORECAST": "solar_forecast_kw",
+}
+MODEL_NAME = "real_data_model"
+MODEL_VERSION = "1.0"
+
+
+@pytest.fixture(scope="function")
+def db_session() -> Generator:
+    """
+    Fixture to set up and tear down a PostgreSQL database session for testing.
+
+    This fixture:
+    - Creates a fresh database schema before each test.
+    - Adds a dummy ML model for test purposes.
+    - Tears down the database session and cleans up resources after each test.
+
+    Returns:
+        Generator: A SQLAlchemy session object.
+    """
+    # Create database engine and tables
+    engine = create_engine(TEST_DB_URL)
```

Comment:
I think you should still do  `with PostgresContainer("postgres:15.5") as postgres:` see other examples here https://github.com/openclimatefix/pv-site-datamodel/blob/main/tests/conftest.py#L26

Reply:
Hi @peterdudfield,I’m still looking into this and will make the necessary changes. I’m sorry if it’s taking too long—would it be okay to give me a few more days to finalize it? I’ll update you as soon as it’s done.